### PR TITLE
Update "sum" example contract

### LINKED ISSUE #59: update sum project

### PR DESCRIPTION
**Pull request by Elli610**
I updated the contract but couldn't verify the front because I couldn't connect it to my local node to broadcast the transactions

closes #59

### CODE REVIEW

**Comment by Ben-Rey** [comment]
Could you add a readme like in the Hello world project ?

File: sum/contracts/package.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
-    "name": "sc-sum",
+    "name": "my-massa-sc",
```
Comment (0xSwapFeeder) [inline sum/contracts/package.json:2]:
```suggestion
    "name": "sum-sc-exemple"
```

**Review by 0xSwapFeeder — APPROVED** [review]
lgtm :)

**Comment by Elli610** [comment]
> Could you add a readme like in the Hello world project ?

Done !

**Review by Ben-Rey — APPROVED** [review]

File: sum/contracts/assembly/contracts/run.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {
+  Address,
+  Storage,
+  call,
+  callerHasWriteAccess,
+} from '@massalabs/massa-as-sdk';
+import { NoArg, Args } from '@massalabs/as-types';
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/run.ts:7]: This should be seen and bloc by the linter in the CI. Can you create an issue to check why the Ci is not working ?

File: sum/contracts/assembly/contracts/run.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
+ */
+export function constructor(sumAddress: StaticArray<u8>): StaticArray<u8> {
+  // This line is important. It ensures that this function can't be called in the future.
+  // If you remove this check, someone could call your constructor function and reset your smart contract.
+  if (!callerHasWriteAccess()) {
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/run.ts:17]: You should use isDeploying function instead

File: sum/contracts/assembly/contracts/run.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  callerHasWriteAccess,
+} from '@massalabs/massa-as-sdk';
+import { NoArg, Args } from '@massalabs/as-types';
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
+ */
+export function constructor(sumAddress: StaticArray<u8>): StaticArray<u8> {
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/run.ts:14]: should return void

File: sum/contracts/assembly/contracts/sum.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { generateEvent, Storage } from '@massalabs/massa-as-sdk';
+import { Args, stringToBytes, i32ToBytes } from '@massalabs/as-types';
+
+/**
+ * @param a - first i32
+ * @param b - second i32
+ * @returns Sum of a and b
+ */
+function add(a: u32, b: u32): u32 {
+  return a + b;
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/sum.ts:10]: You should use safe math...

File: sum/contracts/package.json (hunk 1 above)
Comment (gregLibert) [reply sum/contracts/package.json:2]: I'm still seeing my-massa-sc

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: sum/contracts/assembly/contracts/run.ts (hunk 2 above)
Comment (Elli610) [reply sum/contracts/assembly/contracts/run.ts:7]: Done

File: sum/contracts/assembly/contracts/run.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Address, Storage, call } from '@massalabs/massa-as-sdk';
+import { Args } from '@massalabs/as-types';
+import { isDeployingContract } from '@massalabs/massa-as-sdk/assembly/std/context';
+
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/run.ts:9]: No true anymore

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: sum/contracts/assembly/contracts/run.ts (hunk 6 above)
Comment (gregLibert) [reply sum/contracts/assembly/contracts/run.ts]: Not fixed, your function is having a sumAddress argument

File: sum/contracts/assembly/contracts/run.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  );
+  main([]);
+  return;
+}
+
+/**
+ * Test the 'sum' function of the 'sum' contract by calling it with 2 i32.
+ *
+ * @param _ - not used
+ * @returns empty array
```
Comment (gregLibert) [inline sum/contracts/assembly/contracts/run.ts:29]:
```suggestion
```

File: sum/contracts/src/deploy.ts [hunk 8]
```diff
@@ -23,35 +24,39 @@ const __filename = fileURLToPath(import.meta.url);
...
     publicApi,
     deployerAccount,
     [
       {
-        data: readFileSync(path.join(__dirname, 'build', 'main.wasm')),
-        coins: 0,
-        args: new Args(),
+        data: readFileSync(path.join(__dirname, 'build', 'sum.wasm')),
+        coins: fromMAS(0.5),
+        args: new Args([]),
```
Comment (gregLibert) [inline sum/contracts/src/deploy.ts:34]: You should use noArgs instead

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: sum/contracts/src/deploy.ts (hunk 8 above)
Comment (Elli610) [reply sum/contracts/src/deploy.ts:34]: I can't, NoArg is from as-types. If we want to do so, we should implement it in massa-web3

**Review by gregLibert — APPROVED** [review]

**Pull request merged**